File: core/repository/deployment_causality_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from core.repository.deployment_semantics_engine import analyze_deployment_semantics


def model_deployment_causality(files: List[str]) -> Dict[str, Any]:
    deploy = analyze_deployment_semantics(files)
    causal = [
        {"artifact": a, "causes": "deploy"}
        for a in deploy.get("deployment_artifacts", [])[:30]
    ]
    return {"causal": causal, "semantics": deploy.get("semantics"), "evidence": deploy.get("infra", {}).get("evidence", [])}

File: src/repository/deploymentCausalityEngine.ts
/**
 * Converted from Python: core/repository/deployment_causality_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { analyzeDeploymentSemantics } from "./deploymentSemanticsEngine.js";

export function modelDeploymentCausality(files: any): any {
  var deploy: any = analyzeDeploymentSemantics(files);
  var causal: any = py.iter(py.slice(py.get(deploy, "deployment_artifacts", []), null, 30)).map((a: any) => ({"artifact": a, "causes": "deploy"}));
  return {"causal": causal, "semantics": py.get(deploy, "semantics"), "evidence": py.get(py.get(deploy, "infra", {}), "evidence", [])};
}
export { analyzeDeploymentSemantics };
